modules/firewall-rules/main.tf

```
@@ -14,8 +14,14 @@
  * limitations under the License.
  */
 
+locals {
+  ingress_rules = { for rules in var.ingress_rules : rules.name => merge(rules, { direction = "INGRESS" }) }
+  egress_rules  = { for rules in var.egress_rules : rules.name => merge(rules, { direction = "EGRESS" }) }
```

nit: as I was reading it i got confused for a second why all "rules" when it's being applied to each rule.

```suggestion
  egress_rules  = { for rule in var.egress_rules : rule.name => merge(rule, { direction = "EGRESS" }) }
```